storage/postgres/utils.go

```diff
@@ -75,17 +75,20 @@ func (t *tupleIterator) toArray(opts storage.PaginationOptions) ([]*openfgapb.Tu
 			}
 			return nil, nil, err
 		}
-		ulid = tupleRecord.ulid
 		res = append(res, tupleRecord.asTuple())
 	}
 
-	// Check if we are at the end of the iterator. If we are then we do not need to return a continuation token. This is
-	// why we have LIMIT+1 in the query.
-	if _, err := t.next(); errors.Is(err, storage.TupleIteratorDone) {
-		return res, nil, nil
+	// Check if we are at the end of the iterator. If we are then we do not need to return a continuation token.
+	// This is why we have LIMIT+1 in the query.
+	tupleRecord, err := t.next()
+	if err != nil {
+		if errors.Is(err, storage.TupleIteratorDone) {
+			return res, nil, nil
+		}
+		return nil, nil, err
 	}
 
-	contToken, err := json.Marshal(newContToken(ulid, ""))
+	contToken, err := json.Marshal(newContToken(tupleRecord.ulid, ""))
```

Comment:
Moved picking up the ulid to here so that all our queries look the same. On one of our queries we had `>` instead of `>=` because of this discrepancy. This makes it a bit clearer.